File: app/src/main/java/com/king/frame/base/BaseViewHolder.java
package com.king.frame.base;

import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * @author king
 * @date 2017/3/2 09:56
 */

public abstract class BaseViewHolder extends RecyclerView.ViewHolder{

    public View itemView;

    public BaseViewHolder(View itemView) {
        super(itemView);
        this.itemView = itemView;
        initView();

    }

    public abstract void initView();

    public abstract void bindViewData(int pos);
}

File: app/src/main/java/com/king/frame/base/BaseViewHolder.kt
package com.king.frame.base

import android.view.View

import androidx.recyclerview.widget.RecyclerView

/**
 * RecyclerView通用ViewHolder
 *
 * @author king
 * @date 2019-11-27 11:56
 */

abstract class BaseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    init {
        initView()
    }

    abstract fun initView()

    abstract fun bindViewData(pos: Int)
}
